ТЕМА УСПЕВАЕМОСТЬ СТУДЕНТОВ В ОКСФОРДСКОМ УНИВЕРСИТЕТЕ

In [3]:
import pandas as pd

In [7]:
df = pd.read_csv('student_perfomance.csv')
pd.set_option('display.max_columns', None)
df

Student_ID First_Name Last_Name                       Email  Gender  Age  \
0         S1000       Omar  Williams     student0@university.com  Female   22   
1         S1001      Maria     Brown     student1@university.com    Male   18   
2         S1002      Ahmed     Jones     student2@university.com    Male   24   
3         S1003       Omar  Williams     student3@university.com  Female   24   
4         S1004       John     Smith     student4@university.com  Female   23   
...         ...        ...       ...                         ...     ...  ...   
4995      S5995      Ahmed     Jones  student4995@university.com    Male   19   
4996      S5996       Emma     Brown  student4996@university.com    Male   19   
4997      S5997       John     Brown  student4997@university.com  Female   24   
4998      S5998       Sara     Davis  student4998@university.com    Male   23   
4999      S5999      Maria     Brown  student4999@university.com  Female   21   

       Department  Attendance (%)  Midterm_Score  Final_Score  \
0     Mathematics           97.36          40.61        59.61   
1        Business           97.71          57.27        74.00   
2     Engineering           99.52          41.84        63.85   
3     Engineering           90.38          45.65        44.44   
4              CS           59.41          53.13        61.77   
...           ...             ...            ...          ...   
4995           CS           96.09          45.86        45.63   
4996  Mathematics           96.30          78.67        47.42   
4997           CS           89.94          50.84        65.83   
4998  Engineering           99.09          55.44        57.45   
4999  Mathematics           90.33          93.29        88.42   

      Assignments_Avg  Quizzes_Avg  Participation_Score  Projects_Score  \
0               73.69        53.17                 73.4           62.84   
1               74.23        98.23                 88.0           98.23   
2               85.85        50.00                  4.7           91.22   
3               68.10        66.27                  4.2           55.48   
4               67.66        83.98                 64.3           87.43   
...               ...          ...                  ...             ...   
4995            74.13        81.53                  9.8           51.66   
4996            57.89        85.96                 47.8           52.33   
4997            84.69        82.27                  7.5           97.89   
4998            78.78        55.96                 62.9           84.74   
4999            87.06        65.22                 64.0           66.23   

      Total_Score Grade  Study_Hours_per_Week Extracurricular_Activities  \
0         59.8865     F                  10.3                        Yes   
1         81.9170     B                  27.1                         No   
2         67.7170     D                  12.4                        Yes   
3         51.6535     F                  25.5                         No   
4         71.4030     C                  13.3                        Yes   
...           ...   ...                   ...                        ...   
4995      53.5470     F                   7.8                         No   
4996      59.0240     F                  24.2                         No   
4997      74.7560     C                   7.6                         No   
4998      68.6585     D                  14.5                        Yes   
4999      78.7485     C                  29.9                        Yes   

     Internet_Access_at_Home Parent_Education_Level Family_Income_Level  \
0                         No               Master's              Medium   
1                         No            High School                 Low   
2                         No            High School                 Low   
3                        Yes            High School                 Low   
4                         No               Master's           

In [12]:
df.columns

Index(['Student_ID', 'First_Name', 'Last_Name', 'Email', 'Gender', 'Age',
       'Department', 'Attendance (%)', 'Midterm_Score', 'Final_Score',
       'Assignments_Avg', 'Quizzes_Avg', 'Participation_Score',
       'Projects_Score', 'Total_Score', 'Grade', 'Study_Hours_per_Week',
       'Extracurricular_Activities', 'Internet_Access_at_Home',
       'Parent_Education_Level', 'Family_Income_Level', 'Stress_Level (1-10)',
       'Sleep_Hours_per_Night'],
      dtype='object')

1. Student_ID - Уникальный идентификатор студента.
2. First_Name - Имя студента
3. Last_Name - Фамилия студента
4. Email - Электронная почта
5. Gender - Пол студента
6. Age - Возраст студента
7. Department - Факультет/отделение
8. Attendance (%) - Посещаемость студента
9. Midterm_Score - Результат промежуточного экзамена
10. Final_Score - Результат финального экзамена
11. Assignments_Avg - Средний балл за задания
12. Quizzes_Avg - Средний балл за тесты
13. Participation_Score - Балл за участие
14. Projects_Score - Оценка за проекты
15. Total_Score - Итоговый балл
16. Grade - Итоговая оценка: ключевой показатель успеваемости
17. Study_Hours_per_Week - Часы учебы в неделю
18. Extracurricular_Activities - Внеучебная активность
19. Internet_Access_at_Home - Доступ к интернету дома
20. Parent_Education_Level - Образование родителей
21. Family_Income_Level - Уровень дохода семьи
22. Stress_Level (1-10) - Уровень стресса
23. Sleep_Hours_per_Night - Часы сна за ночь

1. Гипотеза прямой корреляции посещаемости и оценок
*Студенты с высокой посещаемостью (>90%) имеют значительно более высокие оценки (A/B), чем студенты с низкой посещаемостью (<70%).*

2. Гипотеза влияния времени учебы
Студенты, которые занимаются более 15 часов в неделю, имеют средний балл на 20% выше, чем те, кто занимается менее 5 часов.

3. Гипотеза стрессового воздействия
*Студенты с уровнем стресса 8-10 баллов имеют вероятность получить оценку F в 3 раза выше, чем студенты с стрессом 1-3 балла.*

4. Гипотеза сна и когнитивных функций
*Студенты, спящие менее 6 часов в сутки, показывают на 15% более низкие результаты на экзаменах, чем те, кто спит 7-8 часов.*

5. Гипотеза социально-экономического влияния
Студенты из семей с высоким уровнем дохода и высшим образованием родителей имеют преимущество в успеваемости на 0.5 балла по6. Гипотеза оптимального баланса "учеба-сон-стресс"
*Существует нелинейная зависимость: студенты с умеренной учебной нагрузкой (10-15 часов), средним стрессом (4-6) и нормальным сном (7-8 часов) показывают лучшие результаты, чем те, кто максимально нагружает себя.*

7. Гипотеза "парадокса внеучебной активности"
Студенты, участвующо внеучебной активностиties, несмотря на меньшее время на учебу, имеют более высокую эффективность обучения и лучшие оценки благодаря лучшим навыкам тайм-менеджмеде8иях.

9. Гипотеза "критического порога посещаемости"
*Существует порог ~85% посещаемости, после которого дальнейшее увеличение посещаемости не дает значительного прироста в оценках, а после 95% может даже снижать результаты из-за вм 9зультате.

11. Гипотеза "компенсаторного механизма"
Студенты с ограниченным доступом в интернет компенсируют это более высокой посещаемостью и активностью на занятиях, достигая результатов нетив0ость обучения.*

13. Гипотеза "стрессовой продуктивности"
*Умеренный стресс (4-6 баллов) коррелирует с лучшими результатами, чем полное отсутствие стресса (1-2) или экстр-обр1зная зависимость.*

14. Гипотеза синергии проектной работы и экзаменов
Студенты с высокими баллами за проекты показывают непропорционально высокие результаты на финальных экзаменах, даже если изаданиямьтаты по тес2ам и quizzes средние.

15. Гипотеза "обратного влияния оценок на психологическое спромежуточноме"
Низкие оценки на midterm экзамене приводят к увеличению стресса и уменьшению часов сна, что создает порочный круг и еще больше ухудшает финальные результаты. шкале GPA. поверхностном анализе данных.



